File: php/largest.php

<?php

// find the largest number of input list

$nums = array_slice($argv, 1);

$largest = NULL;

foreach ($nums as $value) {
    if ($largest === NULL || $value > $largest) {
        $largest = $value;
    }
}

echo $largest . "\n";

?>

File: ruby/largest.rb

# find the largest number of input list

nums = ARGV.map { |x| x.to_f }

largest = nil

nums.each {
    |i|
    if largest == nil || i > largest
        largest = i
    end
}

puts largest % 1 == 0 ? largest.to_i : largest
